server/app/services/grid_service_instance_deployer.rb

```diff
@@ -32,12 +33,11 @@ def deploy(node, instance_number, deploy_rev, creds = nil)
   # @param [String] deploy_rev
   def wait_for_service_to_start(node, instance_number, deploy_rev)
     # node/agent has 5 minutes to do it's job
-    Timeout.timeout(300) do
-      sleep 0.5 until self.deployed_service_container_exists?(instance_number, deploy_rev)
-      if self.wait_for_port?
-        container = self.find_service_instance_container(instance_number, deploy_rev)
-        sleep 0.5 until port_responding?(container, self.wait_for_port)
-      end
+    wait!(timeout: 300) { self.deployed_service_container_exists?(instance_number, deploy_rev) }
+
+    if self.wait_for_port?
+      container = self.find_service_instance_container(instance_number, deploy_rev)
+      wait!(timeout: 300) { port_responding?(container, self.wait_for_port) }
```

Comment:
the total timeout will be now 10mins as there are two separate wait loops. Not sure if it will cause any side-effects. I think both of these checks could be combined in one wait block?